openshift-with-appstudio-test/e2e/jvm_build_service_test.go

```diff
@@ -6,331 +6,129 @@ package e2e
-	nodeList, err := nodeClient.List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		ta.Logf(fmt.Sprintf("error listin nodes: %s", err.Error()))
-		return
-	}
-	ta.Logf(fmt.Sprintf("dumpNodes found %d nodes in list, but only logging worker nodes", len(nodeList.Items)))
-	for _, node := range nodeList.Items {
-		_, ok := node.Labels["node-role.kubernetes.io/master"]
-		if ok {
-			continue
-		}
-		if node.Status.Allocatable.Cpu() == nil {
-			ta.Logf(fmt.Sprintf("Node %s does not have allocatable cpu", node.Name))
-			continue
+		labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"build.appstudio.openshift.io/component": ta.appstudioComponent.Name}}
+		listOptions := metav1.ListOptions{
+			LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+			Limit:         1,
 		}
-		if node.Status.Allocatable.Memory() == nil {
-			ta.Logf(fmt.Sprintf("Node %s does not have allocatable mem", node.Name))
-			continue
-		}
-		if node.Status.Capacity.Cpu() == nil {
-			ta.Logf(fmt.Sprintf("Node %s does not have capacity cpu", node.Name))
-			continue
-		}
-		if node.Status.Capacity.Memory() == nil {
-			ta.Logf(fmt.Sprintf("Node %s does not have capacity mem", node.Name))
-			continue
-		}
-		alloccpu := node.Status.Allocatable.Cpu()
-		allocmem := node.Status.Allocatable.Memory()
-		capaccpu := node.Status.Capacity.Cpu()
-		capacmem := node.Status.Capacity.Memory()
-		ta.Logf(fmt.Sprintf("Node %s allocatable CPU %s allocatable mem %s capacity CPU %s capacitymem %s",
-			node.Name,
-			alloccpu.String(),
-			allocmem.String(),
-			capaccpu.String(),
-			capacmem.String()))
-	}
-}
-
-func debugAndFailTest(ta *testArgs, failMsg string) {
-	dumpPods(ta, ta.ns)
-	dumpPods(ta, "jvm-build-service")
-	dumpBadEvents(ta)
-	ta.t.Fatalf(failMsg)
-}
-
-func setup(t *testing.T, ta *testArgs) *testArgs {
-	if ta == nil {
-		ta = &testArgs{
-			t:        t,
-			timeout:  time.Minute * 10,
-			interval: time.Second * 15,
+		err = wait.PollImmediate(ta.interval, ta.timeout, func() (done bool, err error) {
+			prs, err := tektonClient.TektonV1beta1().PipelineRuns(ta.ns).List(context.TODO(), listOptions)
+			if err != nil {
+				return false, nil
+			}
+			if len(prs.Items) > 0 {
+				ta.run = &prs.Items[0]
+				return true, nil
+			}
+			return false, nil
+		})
+		if err != nil {
+			debugAndFailTest(ta, fmt.Sprintf("failure occured when waiting for the pipeline run to get created: %v", err))
 		}
-	}
-	setupClients(ta.t)
-
-	if len(ta.ns) == 0 {
-		ta.ns = generateName(testNamespace)
-		_, err := projectClient.ProjectV1().ProjectRequests().Create(context.Background(), &projectv1.ProjectRequest{
-			ObjectMeta: metav1.ObjectMeta{Name: ta.ns},
-		}, metav1.CreateOptions{})
+	})
 
-		if err != nil {
-			debugAndFailTest(ta, fmt.Sprintf("%#v", err))
+	ta.t.Run("build-container task from component pipelinerun references a correct sidecar image", func(t *testing.T) {
+		ciSidecarImage := os.Getenv("JVM_BUILD_SERVICE_SIDECAR_IMAGE")
```

Comment:
So given the last failed test these were no far off @psturc 
`     utils.go:131: the build-container task from component pipelinerun doesn't contain correct sidecar image. expected: registry.build01.ci.openshift.org/ci-op-6tldjcng/pipeline@sha256:43ee9d375e2ad881c63b85a214c91d5d57b4f6fa1969b5991c602b88c404f5cc, actual: registry.build01.ci.openshift.org/ci-op-6tldjcng/pipeline `

the same because the SHA is stripped from the openshift/release env var at https://github.com/redhat-appstudio/jvm-build-service/blob/85aa504011cc5a7e458f65dc35ca317312a4eb00/.ci/common-ci-settings-and-functions.sh#L47

So you need to use https://github.com/redhat-appstudio/jvm-build-service/blob/85aa504011cc5a7e458f65dc35ca317312a4eb00/.ci/common-ci-settings-and-functions.sh#L48-L49

to build the string to compare with @psturc

Reply:
not sure why the analyzer check test did not run @psturc but I would imagine it needs the same adjustment

Reply:
oh ... too much multitasking ... is it the bundle image that is missing the SHA at the end .. if so, then maybe the override is incorrect ?

Reply:
@gabemontero so as we discussed I updated the [common-ci....sh](https://github.com/redhat-appstudio/jvm-build-service/pull/161/commits/607e71dddc36a443b05b53e9bc5323bd1dd5a04e) script a bit to avoid stripping the original env vars coming from openshift/release. This should fix the issue. I hope I didn't miss a place where we rely on the former values of those env vars and it doesn't result in some other failure. We'll see

Reply:
those env vars are referenced in the controller golang code @psturc 
```
./pkg/reconciler/dependencybuild/dependencybuild.go:	image := os.Getenv("JVM_BUILD_SERVICE_SIDECAR_IMAGE")
./pkg/reconciler/dependencybuild/dependencybuild.go:	image := os.Getenv("JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE")
./pkg/reconciler/artifactbuild/artifactbuild.go:	image := os.Getenv("JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE")
```

but of course they are not affected by the machinations in https://github.com/redhat-appstudio/jvm-build-service/pull/161/commits/607e71dddc36a443b05b53e9bc5323bd1dd5a04e

similar, some of the overlay / `make dev` yaml files reference them, but again, the should be unaffected by https://github.com/redhat-appstudio/jvm-build-service/pull/161/commits/607e71dddc36a443b05b53e9bc5323bd1dd5a04e

Reply:
otherwise the proof should be in these tests here ... the image is right in the sidecar / analyzer or it is not right :-)